src/Chapter10.Tests/Listing10.23.RegisteringAFinalizerWithProcessExit.Tests.cs

```diff
@@ -1,41 +1,39 @@
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 using System;
-using System.Diagnostics;
 using System.IO;
-using Chapter10.Tests.PowerShellTestsUtilities;
+using AddisonWesley.Michaelis.EssentialCSharp.Shared.Tests;
+
 using System.Text.RegularExpressions;
 
 namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter10.Listing10_23.Tests
 {
     [TestClass]
-
     public class DisposeTests
     {
+        public TestContext TestContext { get; set; } = null!; // Auto-initialized by MSTest.
 
-        static string Ps1Path { get; } = Path.GetFullPath("../../../../Chapter10/Listing10.23.RegisteringAFinalizerWithProcessExit.ps1", Environment.CurrentDirectory);
-        static bool PowerShellNotAvailable = PowerShellTestsUtilities.PowerShellNotInstalled();
-        static string PowershellEnvironmentVariableName { get; set; } = "powershell";
+        static string Ps1DirectoryPath { get; } =
+            Path.GetFullPath(Path.Join("..", "..", "..", "..", "Chapter10"), Environment.CurrentDirectory);
```

Comment:
I typically hate relative paths like this as they make assumptions on the working directory. We have a solution in the other repo that instead of counting a number of parent directories, it instead looks up through the directories for the ".git" folder to find the repo root and builds relative paths based on that.

Reply:
https://www.nuget.org/packages/IntelliTect.Multitool/#readme-body-tab

Reply:
You are right.  I eliminated the problem in the PowerShell script but not the C# code. @BenjaminMichaelis has been suggesting I use the common library for this but I was reluctant to take an additional library dependency, even in a test. However, your point about working directory is significant.

Reply:
Leveraged Multitool as @BenjaminMichaelis  suggested.